packages/kit/src/core/dev/index.js

```diff
@@ -475,43 +322,113 @@ async function create_plugin(config, dir, https, get_manifest) {
 											)
 											.join('\n\n')}
 									`;
-								}
-							}
+										}
+									}
 
-							return rendered;
-						},
-						trailing_slash: config.kit.trailingSlash
+									return rendered;
+								},
+								trailing_slash: config.kit.trailingSlash
+							}
+						);
+
+						if (rendered) {
+							res.writeHead(rendered.status, rendered.headers);
+							if (rendered.body) res.write(rendered.body);
+							res.end();
+						} else {
+							not_found(res);
+						}
+					} catch (e) {
+						const error = coalesce_to_error(e);
+						vite.ssrFixStacktrace(error);
+						res.statusCode = 500;
+						res.end(error.stack);
 					}
-				);
-
-				if (rendered) {
-					res.writeHead(rendered.status, rendered.headers);
-					if (rendered.body) res.write(rendered.body);
-					res.end();
-				} else {
-					not_found(res);
-				}
-			} catch (e) {
-				const error = coalesce_to_error(e);
-				vite.ssrFixStacktrace(error);
-				res.statusCode = 500;
-				res.end(error.stack);
+				});
+			};
+		}
+	};
+
+	/** @type {[any, string[]]} */
+	const [merged_config, conflicts] = deep_merge(modified_vite_config, {
+		configFile: false,
+		root: cwd,
+		resolve: {
+			alias: {
+				$app: path.resolve(`${dir}/runtime/app`),
+				$lib: config.kit.files.lib
 			}
-		};
+		},
+		plugins: [
+			svelte({
+				extensions: config.extensions,
+				emitCss: !config.kit.amp,
+				compilerOptions: {
+					hydratable: !!config.kit.hydrate
+				}
+			}),
+			kit_plugin
+		],
+		publicDir: config.kit.files.assets,
+		base: '/'
+	});
+
+	print_config_conflicts(conflicts, 'kit.vite.');
+
+	// optional config from command-line flags
+	// these should take precedence, but not print conflict warnings
+	if (host) {
+		merged_config.server.host = host;
+	}
+
+	// if https is already enabled then do nothing. it could be an object and we
+	// don't want to overwrite with a boolean
+	if (https && !merged_config.server.https) {
+		merged_config.server.https = https;
+	}
+
+	if (port) {
+		merged_config.server.port = port;
 	}
 
+	const server = await vite.createServer(merged_config);
+	await server.listen(port);
+
+	const address_info = /** @type {import('net').AddressInfo} */ (
+		/** @type {import('http').Server} */ (server.httpServer).address()
+	);
+
 	return {
-		name: 'vite-plugin-svelte-kit',
-		/**
-		 * @param {import('vite').ViteDevServer} vite
-		 */
-		configureServer(vite) {
-			return () => {
-				remove_html_middlewares(vite.middlewares);
-				vite.middlewares.use(create_kit_middleware(vite));
-			};
-		}
+		address_info,
+		server_config: vite_config.server,
+		allow,
+		cwd,
```

Comment:
I'm not sure that `cwd` or `close` are used?

Reply:
`close` is (by the tests). i'm still working on this PR, it's probably not worth reviewing until it's marked ready — more changes to come